File: c/list-applications.c
/*
 * List the applications registered on at-spi.
 */

#include <atspi/atspi.h>
#include <stdlib.h>
#include <unistd.h>
#include <string.h>


int main(int argc, gchar **argv)
{
  gint i;
  AtspiAccessible *desktop = NULL;
  AtspiAccessible *app = NULL;

  atspi_init ();

  desktop = atspi_get_desktop (0);
  for (i = 0; i < atspi_accessible_get_child_count (desktop, NULL); i++) {
    app = atspi_accessible_get_child_at_index (desktop, i, NULL);

    g_print ("(Index, application, application_child_count)=(%d,%s,%d)\n",
             i, atspi_accessible_get_name (app, NULL), atspi_accessible_get_child_count (app, NULL));
    g_object_unref (app);
  }

  return 1;
}

File: javascript/list-applications.js
/*
 * Usage:
 *    gjs list-applications.js
 *
 * List the applications registered on at-spi
 *
 */

const Atspi = imports.gi.Atspi;

let desktop;
let app;

Atspi.init();
desktop = Atspi.get_desktop(0);
for (let i = 0; i < desktop.get_child_count(); i++) {
    let app = desktop.get_child_at_index (i);

    print ("(Index, application, application_child_count)=("+
           i + "," + app.get_name() + "," + app.get_child_count()+")");
}
